httpsender/Alert on HTTP Response Code Errors.js

```diff
@@ -0,0 +1,50 @@
+// A script which will raise alerts based on HTTP Response codes
+// By default it will raise 'Info' level alerts for Client Errors (4xx) and 'Low' Level alerts for Server Errors (5xx)
+// But it can be easily changed.
+
+// The following handles differences in printing between Java 7's Rhino JS engine
+// and Java 8's Nashorn JS engine
+if (typeof println == 'undefined') this.println = print;
+
+pluginid = 100000	// https://github.com/zaproxy/zaproxy/blob/develop/src/doc/scanners.md
+
+function sendingRequest(msg, initiator, helper) {
+	// Nothing to do
+}
+
+function responseReceived(msg, initiator, helper) {
+	var extensionAlert = org.parosproxy.paros.control.Control.getSingleton().getExtensionLoader().getExtension(
+		org.zaproxy.zap.extension.alert.ExtensionAlert.class)
+	if (extensionAlert != null) {
+		var code = msg.getResponseHeader().getStatusCode()
+		if (code < 400 || code >= 600) {
+			// Do nothing
+		} else {
+			var risk = 0	// Info
+			var title = "A Client Error response code was returned by the server"
+			if (code >= 500) {
+				// Server error
+				risk = 1	// Low
+				title = "A Server Error response code was returned by the server"
+			}
+			// CONFIDENCE_HIGH = 3 (we can be pretty sure we're right)
+			var alert = new org.parosproxy.paros.core.scanner.Alert(pluginid, risk, 3, title)
+			var ref = msg.getHistoryRef()
+			if (ref == null) {
+				ref = new org.parosproxy.paros.model.HistoryReference(
+					org.parosproxy.paros.model.Model.getSingleton().getSession(), initiator, msg)
```

Comment:
Should it use other value for the history type? The ID of the initiators do not match the history types (should map the initiator ID to appropriate history type?).